runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -3878,6 +3878,25 @@ class MM_CopyForwardSchemeRootClearer : public MM_RootScanner
 		}
 	}
 
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+	virtual void doDoubleMappedObjectSlot(J9Object *objectPtr, struct J9PortVmemIdentifier *identifier) {
+		MM_EnvironmentVLHGC *env = MM_EnvironmentVLHGC::getEnvironment(_env);
+		env->_copyForwardStats._doubleMappedArrayletsCandidates += 1;
+		if (!_copyForwardScheme->isLiveObject(objectPtr)) {
+			Assert_MM_true(_copyForwardScheme->isObjectInEvacuateMemory(objectPtr));
+			MM_ScavengerForwardedHeader forwardedHeader(objectPtr);
+			objectPtr = forwardedHeader.getForwardedObject();
+			if (NULL == objectPtr) {
+				Assert_MM_mustBeClass(forwardedHeader.getPreservedClass());
+				env->_copyForwardStats._doubleMappedArrayletsCleared += 1;
+				PORT_ACCESS_FROM_ENVIRONMENT(_env);
+				int result = j9vmem_free_memory(identifier->address, identifier->size, identifier);
```

Comment:
I don't think `int` is the right return type for `j9vmem_free_memory`.  Can you validate what this should be?

Reply:
It's an `int32_t` but I'll get rid of `int result` since it won't be used anywhere